components/Answer/Answer.tsx

```diff
@@ -1,18 +1,78 @@
-import React, { useEffect, useState } from "react";
-import styles from "./answer.module.css";
+import React, { useState } from 'react';
+import { IconThumbUp, IconThumbDown } from '@tabler/icons-react';
 
+import { getSessionId } from '@/utils';
+
+import './answer.module.css';
+import styles from './answer.module.css';
+
+const scheme = process.env.NODE_ENV === 'production' ? 'https://' : 'http://';
+const BACKEND_URL = scheme + process.env.NEXT_PUBLIC_BACKEND_URL;
 interface AnswerProps {
   text: string;
+  query: string;
+  answered: boolean;
 }
 
-export const Answer: React.FC<AnswerProps> = ({ text }) => {
-  const paragraphs = text.split('\n').map((paragraph, index) => (
-    <p key={index}>{paragraph}</p>
-  ));
+export const Answer: React.FC<AnswerProps> = ({ text, query, answered }) => {
+  const paragraphs = text
+    .split('\n')
+    .map((paragraph, index) => <p key={index}>{paragraph}</p>);
+
+  const [votingVisible, setVotingVisible] = useState<boolean>(true);
+  const [feedbackMsg, setFeedbackMsg] = useState<string>('Was this answer useful?');
+
+  const handleVote = async (vote: number) => {
+    const session_id = getSessionId();
+
+    const response = await fetch(`${BACKEND_URL}/vote`, {
+      method: 'POST',
+      body: JSON.stringify({
+        vote: vote,
+        session_id: session_id,
+        question: query,
+      }),
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    });
+
+    if (!response.ok) {
+      /* Handle */
+      alert('Voting error');
+    } else {
+      setVotingVisible(false);
+      setFeedbackMsg('Thank you for your vote!');
```

Comment:
Added a bit of user feedback here:

![Screenshot 2023-05-26 at 09 00 39](https://github.com/raphaelmerx/imigrasi-nextjs/assets/5139869/45e8a6ea-9864-4a81-8856-94e66194db24)